torch/distributed/_composable/fsdp/_fsdp_param_group.py

```diff
@@ -28,9 +36,141 @@ def __init__(
         self.mesh_info = mesh_info
         self.device = device
         self._training_state = TrainingState.IDLE
+        # Group's sharded state always matches its parameters' sharded states
         self._sharded_state = ShardedState.SHARDED
+        self._init_mp_dtypes()
         self._module_fqn: Optional[str] = None  # prefixed from root module
 
+        # - Communication and communication/computation overlap
+        default_stream = torch.cuda.current_stream()
+        self.default_stream: torch.cuda.Stream = default_stream
+        self.all_gather_copy_in_stream: torch.cuda.Stream = default_stream
+        self.all_gather_stream: torch.cuda.Stream = default_stream
```

Comment:
why we by default make these all default stream? can we just make them Nones and let root state shared state assign the correct streams to all param groups?

Reply:
It is for type checking :)

Having something be `Optional[T]` is a pain for mypy.

Reply:
Ah, sorry -- I also refactor this in a later PR: https://github.com/pytorch/pytorch/pull/118004